### DISCUSSION TITLE
Packages and their version

### DISCUSSION THREAD

**Discussion by bcai0411** [SSCHAcode/python-sscha#367] [Q&A] [answered]
Dear all

I encountered a "TypeError: float() argument must be a string or a real number, not 'Atom'" error during my calculation and the SnTe_ToyModel example. I also found the same error in problem #184. After checking, I found that my spglib version is 2.2. It seems there might be version compatibility issues with other packages, such as cellstructure and ase. Could you please list all your packages along with their versions?

**Comment by bcai0411** [comment]
I also found that it cannot read the lattice correctly when the lattice vectors are [[1, 0, 0], [0, 1, 0], [0, 0, 1]]. In this case, the three lattice vectors are incorrectly combined into a single vector [1, 1, 1].

**Comment by mesonepigreco** [comment]
Hi, this seems strange. Could you please copy the full error stack, so we can reconstruct the precise portion of the code raising this issue?

**Reply by DjordjeDangic** [reply]
I think this is related to the spglib version conflict. I usually I get this when I upgrade spglib version. For example:

  File "/home/dangic/WORK/RbH/RbAsH3/R3m/get_third_order/get_third_order.py", line 63, in <module>
    dyn_hessian = ens.get_free_energy_hessian(include_v4 = INCLUDE_V4,
  File "/home/dangic/anaconda3/envs/SSCHA_conda/lib/python3.9/site-packages/sscha/Ensemble.py", line 3718, in get_free_energy_hessian
    qe_sym.SetupFromSPGLIB()
  File "/home/dangic/anaconda3/envs/SSCHA_conda/lib/python3.9/site-packages/cellconstructor/symmetries.py", line 1126, in SetupFromSPGLIB
    spg_syms = spglib.get_symmetry(self.structure.get_ase_atoms(), symprec = self.threshold)
  File "/home/dangic/anaconda3/envs/SSCHA_conda/lib/python3.9/site-packages/spglib/spglib.py", line 621, in get_symmetry
    _, _, _, magmoms = _expand_cell(cell)
  File "/home/dangic/anaconda3/envs/SSCHA_conda/lib/python3.9/site-packages/spglib/spglib.py", line 2065, in _expand_cell
    lattice = np.array(np.transpose(cell[0]), dtype="double", order="C")

**Reply by mesonepigreco** [reply]
Yes, I agree; the "Atom" not being a string or real number error is usually related to SPGLIB > 2.2 dropping the support for ASE Atoms and trying to interpret the ASE structure as a vector. This error is likely due to SPGLIB not correctly downgraded to 2.2 or below. However, verifying without the complete stack of the error @bcai0411 is getting (also related to the lattice) is challenging, especially because @bcai0411 says having installed spglib 2.2. I assume there is a conflict between Python versions and environments.
If no updates follow, I will mark it as solved:

The solution is to check that the SPGLIB version is correctly downgraded to pre-2.2 in the same Python environment as the one executing the SSCHA code

**Reply by mesonepigreco** [reply]
Also cellconstructor and python-sscha should have the same minor version:

**Reply by bcai0411** [reply]
Sorry for late response. Happy new year. Here are my dyn1 to dyn4 files and python code.
[BaTiO3.zip](https://github.com/user-attachments/files/18295944/BaTiO3.zip)

**Reply by bcai0411** [reply]
Here is my packages along with their versions.
[packages-version.zip](https://github.com/user-attachments/files/18295981/packages-version.zip)

**Reply by bcai0411** [reply]
My quantum espresso version is 7.3.1

**Reply by bcai0411** [reply]
I also found "Note: The following floating-point exceptions are signalling: IEEE_DENORMAL" during the sscha calculations. And there is no output information in frequencies.dat.dat file. I am not sure whether it would affect final results.

**Reply by mesonepigreco** [reply, marked as answer]
Hi, this is the typical spglib version error as in #184 
Your python-package specify:
spglib version 2.5
while you must have 2.2 or below (they dropped support for ASE atoms).
Downgrade spglib to version 2.2 and it should work.
We will most likely fix this in a later version of cellconstructor as python-sscha by writing a wrapper, however, it is used quite diffusely in the code right now. 

![image](https://github.com/user-attachments/assets/9214ab92-e411-47fb-9cda-2999fe4f154c)
